### PR TITLE
better curve drawing and tb smoothing

### PR DESCRIPTION
**Pull request by hnyu**
This PR improves the current curve drawing by: 

1. Avoid using TB's SIZE_GUIDANCE["scalars"] because when showing training curves we have far more points before the first summary_interval (if summary_interval is very big). Letting TB take a few points will result in very low curve resolution after the first summary_interval because TB will evenly downsample the points. Now we just load all points and only downsample after fitting them to a function.
2. Add TB's smoothing function which is an EMA style smoothing.
3. Some minor visual adjustment for the plotting function.

### CODE REVIEW

File: alf/utils/plot_tb_curves.py [hunk 1]
```diff
@@ -393,27 +432,40 @@ def _clip_y(y):
             return np.clip(y, y_clipping[0],
                            y_clipping[1]) if y_clipping else y
 
+        if not isinstance(linestyle, list):
+            linestyle = [linestyle] * len(mean_curves)
```
Comment (Haichao-Zhang) [inline alf/utils/plot_tb_curves.py:438]: Why need to do this conversion to the style string? Is there a use case where user can specify a list of style strings that has the same length as the curve?

Comment (hnyu) [reply alf/utils/plot_tb_curves.py:438]: > Why need to do this conversion to the style string? Is there a use case where user can specify a list of style strings that has the same length as the curve?

If you want to mix dashed line styles with solid lines styles for examples (See my paper's curves)

Comment (Haichao-Zhang) [reply alf/utils/plot_tb_curves.py:438]: This needs to be documented somewhere. Currently the docstr ``linestyle (str)`` is not updated.

Comment (hnyu) [reply alf/utils/plot_tb_curves.py:438]: updated

**Review by Haichao-Zhang — APPROVED** [review]

**Pull request merged**